### PR TITLE
Fix remote log in azure storage blob displays in one line

### PR DESCRIPTION
**Pull request by heeroyuy925**
related: https://github.com/apache/airflow/issues/14312

To solve the display issue on UI when using wasb_task_handler.py to store remote log on Azure blob storage

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Comment by flolas** [comment]
Happened after upgrading Blob 2.1v to 12v in Azure Provider #12188. Hope we can have in a release soon.

**Comment by kurtqq** [comment]
probably a good idea to tag the creator of the PR that cause it
@ephraimbuddy

**Comment by ephraimbuddy** [comment]
@kurtqq Thanks for the mention. I think what we should change is replace `readall` in hook.read_file with `content_as_text(max_concurrency=1, encoding='UTF-8')`.  
https://github.com/apache/airflow/blob/c281979982c36f16c4c346c996a0c8d6ca7c630d/airflow/providers/microsoft/azure/hooks/wasb.py#L247
I'm taking a look.
https://docs.microsoft.com/en-us/azure/developer/python/sdk/storage/azure-storage-blob/azure.storage.blob.storagestreamdownloader?view=storage-py-v12#methods

**Comment by ephraimbuddy** [comment]
@heeroyuy925 I have confirmed that we should use `context_as_text` method instead of `readall()` can you update the PR with it? Your solution is Ok but I'm thinking this is less hacky

**Comment by heeroyuy925** [comment]
Thank you for your help @ephraimbuddy I have updated the code as you suggested

File: airflow/providers/microsoft/azure/hooks/wasb.py [hunk 1]
```diff
@@ -244,7 +244,7 @@ def read_file(self, container_name: str, blob_name: str, **kwargs):
         :param kwargs: Optional keyword arguments that `BlobClient.download_blob` takes.
         :type kwargs: object
         """
-        return self.download(container_name, blob_name, **kwargs).readall()
+        return self.download(container_name, blob_name, **kwargs).content_as_text()
```
Comment (ephraimbuddy) [inline airflow/providers/microsoft/azure/hooks/wasb.py:247]:
```suggestion
        return self.download(container_name, blob_name, **kwargs).content_as_text(max_concurrency=1, encoding='utf-8')
```

**Review by ephraimbuddy — CHANGES_REQUESTED** [review]

File: airflow/providers/microsoft/azure/hooks/wasb.py (hunk 1 above)
Comment (heeroyuy925) [reply airflow/providers/microsoft/azure/hooks/wasb.py:247]: I check the Microsoft document you gave above and find these two parameters are default value.
Ok, I can add this as you suggested

Comment (ephraimbuddy) [reply airflow/providers/microsoft/azure/hooks/wasb.py:247]: Hey, seen it. Please don't add

**Review by ephraimbuddy — APPROVED** [review]

File: airflow/providers/microsoft/azure/hooks/wasb.py (hunk 1 above)
Comment (ephraimbuddy) [reply airflow/providers/microsoft/azure/hooks/wasb.py:247]: You have to rebase though

Comment (heeroyuy925) [reply airflow/providers/microsoft/azure/hooks/wasb.py:247]: I have rabased follow the Pull Request Guidelines, please check it

**Pull request merged**